### PR TITLE
Changes to Picking module & fix Picking example 

### PR DESCRIPTION
**Pull request by 1chandu**
- Add separate functions for highlightColor, filterColor in `Picking` module.
- Fix `Picking` example
- Fix Sphere object creation in examples.

### CODE REVIEW

File: src/core/pick-models.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ export default function pickModels(gl, {
         format: gl.RGBA,
         type: gl.UNSIGNED_BYTE});
 
-      const isPicked = color[0] !== 0 || color[1] !== 0 || color[2] !== 0 || color[3] !== 0;
+      const isPicked = color[0] !== 0 || color[1] !== 0 || color[2] !== 0;
```
Comment (1chandu) [inline src/core/pick-models.js]: @ibgreen , when we write picking color to FB , alpha channel is always 1.0, hence removed checking `color[3] !== 0`, verified picking in Instancing example still works.

File: examples/core/picking/app.js [hunk 2]
```diff
@@ -39,16 +43,15 @@ const animationLoop = new AnimationLoop({
       }
     })
     .then(textures => PLANETS.map(
-      (planet, i) => new Sphere({
-        gl,
+      (planet, i) => new Sphere(gl, {
```
Comment (1chandu) [inline examples/core/picking/app.js]: @ibgreen , this fixed Sphere geometry issue, before this change, sphere geometry options (nlat, nlong, etc) are ignored.

File: src/shadertools/modules/picking/README.md [hunk 3]
```diff
@@ -16,28 +16,36 @@ main() {
...
 ```
 main() {
   gl_FragColor = picking_filterHighlight(color);
 
-  gl_FragColor = picking_filterFinal(gl_FragColor);
+  gl_FragColor = picking_filterPickingColor(gl_FragColor);
+}
+```
+
+If you need to apply highlight color you can also just call `picking_filterColor`.
```
Comment (ibgreen) [inline src/shadertools/modules/picking/README.md]: Add this function description to the other doc and delete this doc

File: src/shadertools/modules/picking/picking.js [hunk 4]
```diff
@@ -74,27 +74,32 @@ const float COLOR_SCALE = 1. / 256.;
  * Returns highlight color if this item is selected
  * Note: Should be normally be called before other
  */
-vec4 picking_filterHighlight(vec4 color) {
+vec4 picking_filterHighlightColor(vec4 color) {
   bool selected = bool(picking_vRGBcolor_Aselected.a);
   return selected ? picking_uHighlightColor : color;
 }
 
+vec4 picking_filterPickingColor(vec4 color) {
```
Comment (ibgreen) [inline src/shadertools/modules/picking/picking.js]: Add one line comment header. Keep brief as we don't want to generate too long shader strings.

File: src/shadertools/modules/picking/picking.js [hunk 5]
```diff
@@ -74,27 +74,32 @@ const float COLOR_SCALE = 1. / 256.;
...
+  vec3 pickingColor = picking_vRGBcolor_Aselected.rgb;
+  return picking_uActive ? vec4(pickingColor, 1.0) : color;
+}
+
 /*
  * Returns picking color if picking enabled
```
Comment (ibgreen) [inline src/shadertools/modules/picking/picking.js]: Change to one line comment. We already have separate docs, no need to repeat details here.

File: src/shadertools/modules/picking/picking.js [hunk 6]
```diff
@@ -74,27 +74,32 @@ const float COLOR_SCALE = 1. / 256.;
  * Returns highlight color if this item is selected
  * Note: Should be normally be called before other
```
Comment (ibgreen) [inline src/shadertools/modules/picking/picking.js]: One line comment header, clean up.

File: src/shadertools/modules/picking/picking.js [hunk 7]
```diff
@@ -74,27 +74,32 @@ const float COLOR_SCALE = 1. / 256.;
...
-  return picking_uActive ?
-    vec4(pickingColor, 1.0) :
-    picking_filterHighlight(color);
+  color = picking_filterHighlightColor(color);
+  color = picking_filterPickingColor(color);
+  return color;
 }
 `;
 
+// TODO:  Make changes to doc with above new method and
```
Comment (ibgreen) [inline src/shadertools/modules/picking/picking.js]: This PR handles this TODO about the docs right?

File: docs/api-reference/shadertools/shadertools-picking.md [hunk 8]
```diff
@@ -59,20 +64,23 @@ Sets the color that will be returned by the fragment shader if color based picki
 
 ## Fragment Shader Functions
 
-### picking_filterColor
```
Comment (1chandu) [inline docs/api-reference/shadertools/shadertools-picking.md]: Will remove it, not much advantage.

File: docs/api-reference/geometry/cube.md [hunk 9]
```diff
@@ -52,7 +52,7 @@ The main constructor function for the Sphere class. Use this to create a new Sph
 
 ### Syntax:
 
-  var model = new Sphere(options);
+  var model = new Sphere(gl, options);
```
Comment (1chandu) [inline docs/api-reference/geometry/cube.md]: Done and updated.

**Pull request merged**